posthog-react-native/src/posthog-rn.ts

```diff
@@ -47,31 +71,43 @@ export class PostHog extends PostHogCore {
       this.flush()
     })
 
+    let storagePromise: Promise<void> | undefined
+
     if (this._persistence === 'file') {
-      this._semiAsyncStorage = new SemiAsyncStorage(options?.customAsyncStorage || buildOptimisiticAsyncStorage())
-    }
+      if (options?.customStorage && options?.customAsyncStorage) {
+        console.warn(
+          '[PostHog] was initialised with both customStorage and customAsyncStorage, customStorage will take precedence.'
+        )
+      }
 
-    // Ensure the async storage has been preloaded (this call is cached)
-    const setupAsync = async (): Promise<void> => {
-      if (!this._semiAsyncStorage?.isPreloaded) {
-        await this._semiAsyncStorage?.preloadAsync()
+      if (options?.customStorage) {
+        this._storage = new PostHogRNSyncStorage(options.customStorage)
+      } else {
+        this._storage = new PostHogRNSemiAsyncStorage(options?.customAsyncStorage || buildOptimisiticAsyncStorage())
+        storagePromise = this._storage.preloadPromise
```

Comment:
I would just treat `customStorage` and `customAsyncStorage` the same, and deprecate `customAsyncStorage`. Calling preloadPromise if exists.
`customStorage` typed as generic `PostHogRNStorage` or whatever is generic.

Reply:
Good call.